File: GeneratedEntityReaderClasses/Components/PathFindingGridMarkerComponent.cs
namespace GeneratedNoitaClasses;
public class PathFindingGridMarkerComponent : NoitaComponentBase,  iNoitaType<PathFindingGridMarkerComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt marker_work_flag { get; set; } = new();
	public NoitaFloat marker_offset_x { get; set; } = new();
	public NoitaFloat marker_offset_y { get; set; } = new();
	public NoitaFloat player_marker_radius { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		marker_work_flag.Read(s);
		marker_offset_x.Read(s);
		marker_offset_y.Read(s);
		player_marker_radius.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		marker_work_flag.Write(s);
		marker_offset_x.Write(s);
		marker_offset_y.Write(s);
		player_marker_radius.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/PathFindingGridMarkerComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class PathFindingGridMarkerComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    marker_work_flag:NoitaInt = field(default_factory=NoitaInt)
    marker_offset_x:NoitaFloat = field(default_factory=NoitaFloat)
    marker_offset_y:NoitaFloat = field(default_factory=NoitaFloat)
    player_marker_radius:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.marker_work_flag.Read(s)
        self.marker_offset_x.Read(s)
        self.marker_offset_y.Read(s)
        self.player_marker_radius.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.marker_work_flag.Write(s)
        self.marker_offset_x.Write(s)
        self.marker_offset_y.Write(s)
        self.player_marker_radius.Write(s)

